File: src/clearframe/timecode.py
"""SMPTE-style timecode helpers (non-drop-frame)."""


def seconds_to_tc(seconds: float, fps: float = 24.0, offset_s: float = 3600.0) -> str:
    total_frames = round((seconds + offset_s) * fps)
    frames_per_hour = round(fps * 3600)
    frames_per_minute = round(fps * 60)
    frames_per_second = round(fps)

    hh, rem = divmod(total_frames, frames_per_hour)
    mm, rem = divmod(rem, frames_per_minute)
    ss, ff = divmod(rem, frames_per_second)
    return f"{hh:02d}:{mm:02d}:{ss:02d}:{ff:02d}"

File: webapp/src/timecode.ts
export function tc(seconds: number, fps = 24, offsetS = 3600): string {
  const totalFrames = Math.round((seconds + offsetS) * fps);
  const fph = Math.round(fps * 3600);
  const fpm = Math.round(fps * 60);
  const fpsI = Math.round(fps);
  const hh = Math.floor(totalFrames / fph);
  const mm = Math.floor((totalFrames % fph) / fpm);
  const ss = Math.floor((totalFrames % fpm) / fpsI);
  const ff = totalFrames % fpsI;
  const p = (n: number) => String(n).padStart(2, "0");
  return `${p(hh)}:${p(mm)}:${p(ss)}:${p(ff)}`;
}
